src/main/shared/models/all-stores.ts

```diff
@@ -0,0 +1,5 @@
+import { BulletStore } from '../bullet/bullet-store';
+import { MapStore } from '../map/map-store';
+import { PlayerStore } from '../player/player-store';
+
+export type AllStores = MapStore | PlayerStore | BulletStore;
```

Comment:
Why is this bettern than a generic Store? Looks tedious to update when creating a new store.

Reply:
As in `Store<T>` without a type? so `Store`? 

It is tedious yes, but its more semantically correct, easier to understand in some areas, such as: 

`{ [key: string]: AllStores }`, which is the `dataSource$` typing. This is more accurate and focused than: 

`{ [key: string]: Store<unknown> }`

Also, by using `AllStores`, the codebase and intelligently figure out what stores are available, and will give better IntelliSense

An example is in `dataSource$`:

With the generic type (`{ [key: string]: Store<unknown> }`), this is valid:

```ts
dataSource$.subscribe((source) => source['cool'] as Store<string>)
```

Where as in the focused one, it is not allowed because Store<string> isn't real

Better to help typescript as much as possible :P